File: app/src/main/java/com/iita/akilimo/entities/ProfileInfo.java
package com.iita.akilimo.entities;

import com.iita.akilimo.utils.enums.EnumGender;

import io.objectbox.annotation.Convert;
import io.objectbox.annotation.Entity;
import io.objectbox.annotation.Id;
import io.objectbox.converter.PropertyConverter;

@Entity
public class ProfileInfo {
    @Id
    long id;

    private String deviceID;
    public String userName;
    public String firstName;
    public String lastName;
    public String email;
    public String mobileCode;
    public String fullMobileNumber;
    public String farmName;
    public String fieldDescription;
    public String gender;

    public int selectedGenderIndex;

    
    public boolean sendEmail;

    
    public boolean sendSms;


    public boolean isSendEmail() {
        return this.sendEmail;
    }

    public boolean isSendSms() {
        return this.sendSms;
    }


    public String getNames() {
        return String.format("%s %s", this.firstName, this.lastName);
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getDeviceID() {
        return deviceID;
    }

    public void setDeviceID(String deviceID) {
        this.deviceID = deviceID;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getMobileCode() {
        return mobileCode;
    }

    public void setMobileCode(String mobileCode) {
        this.mobileCode = mobileCode;
    }

    public String getFullMobileNumber() {
        return fullMobileNumber;
    }

    public void setFullMobileNumber(String fullMobileNumber) {
        this.fullMobileNumber = fullMobileNumber;
    }

    public String getFarmName() {
        return farmName;
    }

    public void setFarmName(String farmName) {
        this.farmName = farmName;
    }

    public String getFieldDescription() {
        return fieldDescription;
    }

    public void setFieldDescription(String fieldDescription) {
        this.fieldDescription = fieldDescription;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public int getSelectedGenderIndex() {
        return selectedGenderIndex;
    }

    public void setSelectedGenderIndex(int selectedGenderIndex) {
        this.selectedGenderIndex = selectedGenderIndex;
    }

    public void setSendEmail(boolean sendEmail) {
        this.sendEmail = sendEmail;
    }

    public void setSendSms(boolean sendSms) {
        this.sendSms = sendSms;
    }
}

File: app/src/main/java/com/iita/akilimo/entities/ProfileInfo.kt
package com.iita.akilimo.entities

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey


@Entity(tableName = "profile_info", indices = [Index(value = ["userName"], unique = true)])
open class ProfileInfo {

    @PrimaryKey(autoGenerate = true)
    var profileId: Int? = null
    var deviceToken: String? = null

    var userName: String? = null

    var firstName: String? = null

    var lastName: String? = null
    var email: String? = null
    var mobileCode: String? = null
    var fullMobileNumber: String? = null
    var farmName: String? = null
    var fieldDescription: String? = null

    var gender: String? = null

    var selectedGenderIndex: Int = 0
    var sendEmail: Boolean = false
    var sendSms: Boolean = false

    var language: String? = null
    var countryCode: String? = null
    var countryName: String? = null
    var selectedCountryIndex: Int = 0
    var currency: String? = null

    fun getNames(): String {
        return String.format("%s %s", firstName, lastName)
    }
}